### ISSUE TITLE
RKE2 Provisioning: Generic Machine Driver (Node Pool)

### ISSUE THREAD

**Issue by deniseschannon** [rancher/rancher#32373]
Test Generic Machine Driver as a Node Pool

```
apiVersion: provisioning.cattle.io/v1
kind: Cluster
metadata:
  name: #string
  annotations:
    {}
#    key: string
  labels:
    {}
#    key: string
  namespace: fleet-default
spec:
  cloudCredentialSecretName: <NAME OF CLOUD CREDENTIAL>
  kubernetesVersion: v1.20.5+rke2r1
  rkeConfig:
      nodePools:
#      - cloudCredentialSecretName: string
#        controlPlaneRole: boolean
#        displayName: string
#        etcdRole: boolean
#        hostnamePrefix: string
#        labels:
#          key: string
#        name: string
#        nodeConfig:
#          apiVersion: string
#          fieldPath: string
#          kind: string
#          name: string
#          namespace: string
#          resourceVersion: string
#          uid: string
#        paused: boolean
#        quantity: int
#        rollingUpdate:
#          maxSurge: string
#          maxUnavailable: string
#        taints:
#          - effect: string
#            key: string
#            timeAdded: string
#            value: string
#        workerRole: boolean
```

**Comment by deniseschannon** [comment]
Note: Any of these should have Cloud Credential v2 support automatically.

**Comment by izaac** [comment]
Planning to use OCI for this but it was two known issues with pending validation:

https://github.com/rancher/rancher/issues/33865
https://github.com/rancher/rancher/issues/34302

**Comment by izaac** [comment]
Rancher version: `v2.6-head (09/13/2021)` _6a0a991_
Rancher version: `master-head (09/13/2021)` _452be57_

Enabled Oracle OCI Custom Driver.
Provisioned RKE2 Downstream clusters with different roles pools.

I was able to provision and edit the nodepool sacaling up and down. Deleting and check auto healing worked.
I also checked pools with etcd/Control Plane/worker individual roles. `All` roles and `Master` (etcd/Control Plane) `Worker`.


A separate issue was opened during testing for follow up: https://github.com/rancher/rancher/issues/34713
